src/Sylius/Bundle/PromotionBundle/Validator/CatalogPromotionActionValidator.php

```diff
@@ -21,34 +22,37 @@
 
 final class CatalogPromotionActionValidator extends ConstraintValidator
 {
+    private array $actionTypes;
+
+    private array $actionValidators;
+
+    public function __construct(array $actionTypes, iterable $actionValidators)
+    {
+        $this->actionTypes = $actionTypes;
+        $this->actionValidators = $actionValidators instanceof \Traversable ? iterator_to_array($actionValidators) : $actionValidators;
+    }
+
     public function validate($value, Constraint $constraint): void
     {
         /** @var CatalogPromotionAction $constraint */
         Assert::isInstanceOf($constraint, CatalogPromotionAction::class);
 
         /** @var CatalogPromotionActionInterface $value */
-        if ($value->getType() !== CatalogPromotionActionInterface::TYPE_PERCENTAGE_DISCOUNT) {
+        if (!in_array($value->getType(), $this->actionTypes, true)) {
```

Comment:
It appears that taking into account this snippet:
![image](https://user-images.githubusercontent.com/6213903/143300395-6f872d1b-6cda-40a1-b247-df80c9dad4e8.png)
one may add custom validator for a new action type, yet won't be able to use it, because they would need to override this service definition as we are hardcoded for two choices anyway

Reply:
That's true, it looks the same with validators for scopes. What is your suggestion? The simplest one is probably to have a separate parameter that is an array of these types 🤔

Reply:
The parameter would be a good improvement. However, going deeper what would you say, that these consts would be defined by calculators themselves? Then we could tag them and use their const as a tag attribute and here use another tag iterator. Going further, we could define expectations for static method with the name, but I'm not sure if it makes sense to introduce it day before the end of sprint 
ref. https://symfony.com/doc/current/service_container/tags.html#tagged-services-with-index

Reply:
I'm 👌🏻 for refactoring, but I would be for making it in a separate PR for both actions and scopes